src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -68,12 +67,8 @@ public UniqueStepList getSteps() {
         return steps;
     }
 
-    /**
-     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
-     * if modification is attempted.
-     */
     public Set<Tag> getTags() {
-        return Collections.unmodifiableSet(tags);
+        return tags;
```

Comment:
I think the EditRecipeDescriptor should be changed instead, to use `new HashSet<>(tags)`

Reply:
i thought we only use EditRecipeDescriptor to update the part we need 
but we still have to get the variable itself like tag set/steplist/ingredientlist from the model class first though.